### DISCUSSION TITLE
Properly showing enemy hitpoints

### DISCUSSION THREAD

**Discussion by RobbieNeko** [cataclysmbn/Cataclysm-BN#4127] [General]
As once mentioned in the Discord (then promptly buried):

The current system surrounding seeing enemy health in game is pretty bad. It would be much more helpful if we could see the proper health values of enemies. How are you going to plan around/in combat when you don't even know (at least roughly) how much HP your enemies have left? We know how much damage our weapons will generally do in fairly precise numbers, yet this information is of little use unless we know the HP of the enemy to know roughly how many hits they will need.

However, I can also see that there may be a desire to preserve some of the aspect of (new players at least) not knowing the exact health figures of enemies early on. Hence, this proposal.

As a skill increases, the accuracy of the assessment of the enemy's health will increase alongside it. To use a very rough example, and assuming that we make a rather logical choice of using First Aid as the relevant skill, here's an example (with all three of these being descriptions of the same zombie, just at different levels of detail/accuracy):
- First aid 0: You estimate the zombie has between 50 and 110 max HP.
- First aid 5: You estimate that of the zombie's 80 max HP, about 3/4 is remaining.
- First aid 10: The zombie has 58 HP remaining out of its max of 80. 

Or, in a more condensed format (with thanks to Kheir)
- ??/50-130
- ~60/80
- 57/80

At the start, your character is merely guessing at a range where the true value is somewhere within that range. This range will tighten as you gain levels, until you know definitively the maximum HP. From there, we move on to a descriptor like seen at first aid 5 here, which has an estimation of how much of that known max health is left. Naturally, this would get more and more accurate/precise fractions or percentages until we reach the point of knowing the exact value, as seen in the First Aid 10 example. For the first level of health estimation, it could be a decent idea to make the range slightly asymmetrical so that it is not merely a matter of "take the median to find the true max HP value" 

It may be the case that the exact progression of this information should be changed for various reasons, with me originally using these levels as nice round numbers for the sake of explanation/demonstration. Also, alongside this, it would be a good idea to do similar for the player's own health for similar reasons as to why knowing the actual enemy health is highly desirable. 

Thoughts?

**Comment by NappingOcean** [comment]
Speaking of checking the enemy's HP, here's another idea:
An icon next to the enemy that is visible to the player, indicating the current "woundedness".


IIRC, the current icon for creatures is red/yellow/green (transparent) ! which indicates their hostility towards the player. Similar to this, it would be nice to have a symbol icon that changes color for the "creature's current health" (if it's not injured, it won't show up, presumably).

The symbol could be ♥︎, which is an RPG staple, or ||, which is the symbol used in BN to indicate durability, which would be very appropriate.

This icon HP system would be especially useful when dealing with hordes of enemies.

**Reply by RobbieNeko** [reply]
A heart icon in addition to the numbers could indeed be a good idea for telling at a quick glance which enemies got wounded/already have some wounds

**Reply by NappingOcean** [reply]
(No symbol) --> `||` --> `|\` --> `|.` --> `\.` --> `XX`

  `||` ~~(idk its name. 2 bars symbol?)~~ looks cool since it shows states with not only color, even also its **shape**. This form shows the info much better than the form with color only.
  Furthermore this is used Cataclysm already, so it's friendly to Cata-users.

**Comment by scarf005** [comment]
![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/54838975/756830dc-0b1e-43b7-8e78-aefd3346ab06)

actaully, in debug mode we can see the full hitpoint of enemy. maybe we could improve upon this.
